build_support/check_code_style.sh

```diff
@@ -270,14 +268,12 @@ if $PERFORM_CPPCHECK; then
   fi
 fi
 
-# Verify the CLANG-FORMAT installation. CLANG-FORMAT version 3.6 and only 3.6 is required.
-# The CLANG-FORMAT versions up to and including 3.5 do not support all configuration options required for NEST.
-# Version 3.7 introduced a different formatting. NEST relies on the formatting of version 3.6.
+# Verify the CLANG-FORMAT installation. CLANG-FORMAT version 9 is required.
 if $PERFORM_CLANG_FORMAT; then
   $CLANG_FORMAT -style=./.clang-format ./nest/main.cpp >/dev/null 2>&1 || error_exit "Failed to verify the CLANG-FORMAT installation. Executable: $CLANG_FORMAT"
```

Comment:
With clang-format 9.0.1 on macOS (Apple Silicon), style must be given as string "file", not as file name:
```suggestion
  $CLANG_FORMAT -style=file ./nest/main.cpp >/dev/null 2>&1 || error_exit "Failed to verify the CLANG-FORMAT installation. Executable: $CLANG_FORMAT"
```

This seems consistent with the output of `clang-format-9 --help`:
```
  --style=<string>           - Coding style, currently supports:
                                 LLVM, Google, Chromium, Mozilla, WebKit.

                               Use -style=file to load style configuration from
                               .clang-format file located in one of the parent
                               directories of the source file (or current
                               directory for stdin).
```